src/components/icons/icons.tsx

```diff
@@ -12,114 +11,125 @@ export interface IonIconProps {
-    const proportions = {
-      largeIcon: {
-        inner: 1.5,
-        outer: 2.25,
-      },
-      smallIcon: {
-        inner: 1.75,
-        outer: 2.5,
-      },
-    };
-
-    const iconSize = size >= mdIcon ? 'largeIcon' : 'smallIcon';
-
-    return {
-      innerCircle: proportions[iconSize].inner,
-      outsideCircle: proportions[iconSize].outer,
-    };
+  const outerStylesControl = {
+    double: {
+      color: `${color}1A`,
+      size: `${size * getCircleProportion(size).outsideCircle}px`,
+    },
+    simple: {
+      color: `${color}1A`,
+      size: `${size * 2}px`,
+    },
+    none: defaultContainerStyle,
   };
 
-  const outerContainerStyle = (): { color: string; size: string } => {
-    if (!validateHexColor(color)) {
-      return defaultContainerStyle;
-    }
-
-    const stylesControl = {
-      double: {
-        color: `${color}1A`,
-        size: `${size * getCircleProportion(size).outsideCircle}px`,
-      },
-      simple: {
-        color: `${color}1A`,
-        size: `${size * 2}px`,
-      },
-      none: defaultContainerStyle,
-    };
-
-    return stylesControl[highlight];
+  const innerStylesControl = {
+    double: {
+      color: `${color}40`,
+      size: `${size * getCircleProportion(size).innerCircle}px`,
+    },
+    simple: defaultContainerStyle,
+    none: defaultContainerStyle,
   };
 
-  const innerContainerStyle = (): { color: string; size: string } => {
-    if (!validateHexColor(color)) {
-      return defaultContainerStyle;
-    }
-
-    const stylesControl = {
-      double: {
-        color: `${color}40`,
-        size: `${size * getCircleProportion(size).innerCircle}px`,
-      },
-      simple: defaultContainerStyle,
-      none: defaultContainerStyle,
-    };
+  const iconPath = pathPurify(type);
 
-    return stylesControl[highlight];
-  };
+  const iconContent = (
+    <Icon
+      $color={color}
+      $size={size}
+      data-testid={`ion-icon-${type}`}
+      viewBox='0 0 24 24'
+      dangerouslySetInnerHTML={{
+        __html: iconPath,
+      }}
+    />
+  );
 
-  const iconPath = pathPurify(type);
+  if (!isHex(color)) {
+    return (
+      <IconHighlight
+        $color={defaultContainerStyle.color}
+        $size={defaultContainerStyle.size}
+        data-testid='ion-icon-outer-highlight'
+      >
+        <IconHighlight
+          $color={defaultContainerStyle.color}
+          $size={defaultContainerStyle.size}
+          data-testid='ion-icon-inner-highlight'
+        >
+          {iconContent}
+        </IconHighlight>
+      </IconHighlight>
+    );
+  }
 
   return (
     <IconHighlight
-      $color={outerContainerStyle().color}
-      $size={outerContainerStyle().size}
-      $isHighlighted={highlight !== 'none' && validateHexColor(color)}
-      data-testid={'ion-icon-outer-highlight'}
+      $color={outerStylesControl[highlight].color}
+      $size={outerStylesControl[highlight].size}
+      $isHighlighted={highlight !== 'none' && isHex(color)}
```

Comment:
```suggestion
      $isHighlighted={highlight !== 'none'}
```
isHex(color) will always be true, this is redundant.